front-end/src/state/contexts/stateContext.js

```diff
@@ -0,0 +1,32 @@
+import React, { createContext, useContext, useReducer } from 'react';
+import { productAction } from './actions';
+
+const StateContext = createContext();
+
+export const useState = () => useContext(StateContext);
+
```

Comment:
it's matching with useState so you can give it some other name like useAllStates

Reply:
ohkk that I missed, will surely change the name